transformer_engine/pytorch/triton/cross_entropy.py

```diff
@@ -90,14 +94,16 @@ def cross_entropy_forward(
         world_size=world_size,
         ignore_idx=ignore_idx,
         n_cols=V,
-        n_non_ignore=n_rows,
+        n_non_ignore=n_non_ignore,
```

Comment:
I see that the kernel uses `n_non_ignore` both to access an intermediate buffer and to scale the reduced loss:
https://github.com/NVIDIA/TransformerEngine/blob/8a37042e5cfa9fe0a55c1e6481c6c946bf23637c/transformer_engine/common/triton/cross_entropy.py#L143
https://github.com/NVIDIA/TransformerEngine/blob/8a37042e5cfa9fe0a55c1e6481c6c946bf23637c/transformer_engine/common/triton/cross_entropy.py#L209

Won't this change break accesses to the intermediate buffer? I think we should pass in a GPU buffer for the number of non-ignored entries and we should rename `n_non_ignore` to `n_rows` to be less deceptive.